thrust/detail/vector_base.inl

```diff
@@ -195,6 +195,34 @@ template<typename T, typename Alloc>
   return *this;
 } // end vector_base::operator=()
 
+  template<typename T, typename Alloc>
+    vector_base<T,Alloc>
+      ::vector_base(std::initializer_list<T> il)
+        :m_storage(),
+         m_size(0)
+  {
+    range_init(il.begin(), il.end());
+  } // end vector_base::vector_base()
```

Comment:
I'm lazy, so I'm always a fan of delegating to other constructors when possible. In this case, we can delegate to the iterator pair ctor: https://github.com/NVIDIA/thrust/blob/37a638e54a58a43955281c3c0c76b7627d34f1e6/thrust/detail/vector_base.inl#L309 

```suggestion
  template<typename T, typename Alloc>
    vector_base<T,Alloc>
      ::vector_base(std::initializer_list<T> il) : vector_base(std::cbegin(il), std::cend(il)) {}
```

Reply:
We did not touch those constructors and we should either move all of them to a consistent pattern or leave them as is.